### PR TITLE
apps: add widget for inspecting and changing cluster app versions in the new UI

### PR DESCRIPTION
**Pull request by axbarsan**
Towards https://github.com/giantswarm/giantswarm/issues/18522

This PR adds a new widget for changing and inspecting app versions. With it you can upgrade/downgrade the version of your app or see details about a specific version.

## Preview 

<details>
<summary>Current version pre-selected</summary>

image

</details>

<details>
<summary>Dropdown with current version pre-selected</summary>

image

</details>

<details>
<summary>Trying to upgrade</summary>

image

</details>

<details>
<summary>current pre-release version that is no longer ensured on the management cluster</summary>

image

</details>

### CODE REVIEW

File: src/components/MAPI/apps/utils.ts [hunk 1]
```diff
@@ -530,15 +530,32 @@ export async function updateAppVersion(
...
-    namespace,
-    appName
-  );
+  let app = await applicationv1alpha1.getApp(client, auth, namespace, appName);
   app.spec.version = version;
 
-  return applicationv1alpha1.updateApp(client, auth, app);
+  app = await applicationv1alpha1.updateApp(client, auth, app);
+
+  mutate(
```
Comment (axbarsan) [inline src/components/MAPI/apps/utils.ts:538]: 👏 consistency 👏

File: src/components/UI/Display/MAPI/apps/AppVersionInspectorOption.tsx [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+interface IAppVersionInspectorOptionProps
+  extends React.ComponentPropsWithoutRef<typeof Box> {
+  version?: string;
+  creationDate?: string;
+  upstreamVersion?: string;
+  isSelected?: boolean;
+  isCurrent?: boolean;
+}
+
+const AppVersionInspectorOption: React.FC<IAppVersionInspectorOptionProps> = ({
```
Comment (axbarsan) [inline src/components/UI/Display/MAPI/apps/AppVersionInspectorOption.tsx:26]: This component represents each option in the version select dropdown

File: src/components/UI/Display/MAPI/apps/AppVersionInspectorOption.tsx [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+  upstreamVersion,
+  isSelected,
+  isCurrent,
+  ...props
+}) => {
+  const textWeight = isSelected ? 'bold' : 'normal';
+
+  return (
+    <Box
+      pad={{ horizontal: 'small', vertical: '7px' }}
```
Comment (axbarsan) [inline src/components/UI/Display/MAPI/apps/AppVersionInspectorOption.tsx:38]: Matching the height with the buttons 😅

File: src/components/MAPI/apps/ClusterDetailAppListWidgetVersionInspector.tsx [hunk 4]
```diff
@@ -0,0 +1,323 @@
...
+              from version{' '}
+              <Truncated as='code' numStart={10}>
+                {app?.spec.version ?? ''}
+              </Truncated>{' '}
+              to{' '}
+              <Truncated as='code' numStart={10}>
+                {currentSelectedVersion ?? ''}
+              </Truncated>
+              ?
+            </Text>
```
Comment (axbarsan) [inline src/components/MAPI/apps/ClusterDetailAppListWidgetVersionInspector.tsx:291]: Doesn't get uglier than this

**Review by kuosandys — APPROVED** [review]

**Pull request merged**